code/modules/mob/living/carbon/xenomorph/hive_datum.dm

```diff
@@ -255,60 +270,114 @@
 
 	if(isxenoqueen(X))
 		on_queen_death(X)
+	if(isxenoshrike(X))
+		on_shrike_death(X)
+	if(X == living_xeno_ruler)
+		on_ruler_death(X)
+
 	return TRUE
 
+
 /datum/hive_status/proc/on_xeno_revive(mob/living/carbon/xenomorph/X)
 	dead_xenos -= X
 	add_to_lists(X)
 	return TRUE
 
+
 // ***************************************
-// *********** Queen
+// *********** Ruler
 // ***************************************
 
-// This proc attempts to find a new queen to lead the hive, if there isnt then it will announce her death
```

Comment:
dont delete comments like this

Reply:
The comment is no longer true, the proc behaves differently now.